test/Microsoft.Health.Fhir.Tests.E2E/Rest/BasicAuthTests.cs

```diff
@@ -140,5 +125,54 @@ public async Task WhenUpdatingAResource_GivenAUserWithUpdatePermissions_TheServe
             Assert.NotEqual(createdResource.Meta.VersionId, updatedResource.Meta.VersionId);
             Assert.NotEqual(createdResource.Meta.LastUpdated, updatedResource.Meta.LastUpdated);
         }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task WhenCreatingAResource_GivenAClientWithNoAuthToken_TheServerShouldReturnUnauthorized()
+        {
+            await Client.RunAsClientApplication(TestApplications.InvalidClient);
+
+            FhirException fhirException = await Assert.ThrowsAsync<FhirException>(async () => await Client.CreateAsync(Samples.GetDefaultObservation()));
+            Assert.Equal(UnauthorizedMessage, fhirException.Message);
+            Assert.Equal(HttpStatusCode.Unauthorized, fhirException.StatusCode);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task WhenCreatingAResource_GivenAClientWithInvalidAuthToken_TheServerShouldReturnUnauthorized()
+        {
+            await Client.RunAsClientApplication(TestApplications.InvalidClient);
+            Client.SetBearerToken(Invalidtoken);
+            FhirException fhirException = await Assert.ThrowsAsync<FhirException>(async () => await Client.CreateAsync(Samples.GetDefaultObservation()));
+            Assert.Equal(UnauthorizedMessage, fhirException.Message);
+            Assert.Equal(HttpStatusCode.Unauthorized, fhirException.StatusCode);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task WhenCreatingAResource_GivenAClientWithWrongAudience_TheServerShouldReturnUnauthorized()
+        {
+            await Client.RunAsClientApplication(TestApplications.WrongAudienceClient, AuthenticationScenarios.AUTHWITHWRONGAUDIENCE);
```

Comment:
Since we're only ever passing a different variable as the authenticationscenario once, would it make sense to not use an enum and maybe just pass some sort of flag? 

Just a random thought, not sure if it makes it cleaner or not.